src/shared/parse-helpers.c

```
@@ -196,3 +198,53 @@ int parse_socket_bind_item(
         *port_min = mn;
         return 0;
 }
+
+int open_file_parse(const char *v, OpenFile **ret) {
+        _cleanup_free_ char *path = NULL, *fdname = NULL, *flags = NULL;
+        _cleanup_(open_file_freep) OpenFile *of = NULL;
+        int r;
+
+        assert(v);
+
+        of = new0(OpenFile, 1);
+        if (!of)
+                return log_oom();
+
+        r = extract_many_words(&v, ":", EXTRACT_DONT_COALESCE_SEPARATORS, &of->path, &of->fdname, &flags, NULL);
+        if (r < 0)
+                return r;
+
+        if (r == 0)
+                return -EINVAL;
+
+        if (!path_is_absolute(of->path))
+                return -EINVAL;
+
+        if (isempty(of->fdname)) {
+                r = free_and_strdup(&of->fdname, of->path);
+                if (r < 0)
+                        return r;
+        }
+
+        if (flags && streq(flags, "ro"))
+                of->flags = OPENFILE_RDONLY;
```

So we might only have one option now but we could add more in the future so let's make this slightly more future proof. You should use `extract_word()` here in a loop to loop over all the options (using `,` as the separator and parse each option individually.

To do the option parsing, you should make a string table for the flags using `DEFINE_STRING_TABLE_LOOKUP()`. See `PortableFlags` again for an example.

---

We should also fail if we encounter any unknown options